agent/agentsocket/server_unix.go

```diff
@@ -35,7 +35,17 @@ type Server struct {
 }
 
 // NewServer creates a new agent socket server.
-func NewServer(path string, logger slog.Logger) (*Server, error) {
+func NewServer(logger slog.Logger, opts ...Option) (*Server, error) {
```

Comment:
Fine for now, but heads-up: re-using the same Options for server and client is typically not a good idea. Using (Server)Option and ClientOption may become necessary at some point. (You want to avoid misleading the user by exposing options that are ineffective for the type.)

Reply:
Thanks! Yeah, I agree but its simple enough for now that I think its fine. Will defer splitting them out.